src/components/timeSlotBooking/TimeSlotDetails.tsx

```diff
@@ -153,13 +163,45 @@ export default function TimeSlotDetails({
     );
   };
 
+  const addLocalContact = (data: BasicUserDetails[]) => {
+    const [selectedLocalContact] = data;
+
+    if (selectedLocalContact) {
+      handleSetDirty(true);
+      onSave({
+        ...scheduledEvent,
+        localContact: selectedLocalContact as unknown as User,
```

Comment:
This could cause problems latter, where localContact is accessed as User, but only has BasicUser properties.

Reply:
Let's try to resolve this a bit better.